### ISSUE TITLE
Build failure on main due to test involving RemoveDependency

### ISSUE THREAD

**Issue by timtebeek** [openrewrite/rewrite-gradle-plugin#279]
Build seems to have started failing after this change to RemoveDependency (which could be a coincidence)
https://github.com/openrewrite/rewrite/commit/3bd8271be963c025ce0856072d0f09360e3184cd#diff-1958a62558bad4f9c614d5b4697f627f72c5cc61d1674d04880fdae97df67237

Replicating the same test in openrewrite/rewrite has not effect; this passes:
```java
@Test
void removeFromYaml() {
    rewriteRun(
      spec -> spec.recipeFromYaml(
        """
              type: specs.openrewrite.org/v1beta/recipe
              name: org.openrewrite.test.RemoveJacksonCore
              displayName: Remove jackson-core
              description: Remove jackson-core.
              recipeList:
                - org.openrewrite.gradle.RemoveDependency:
                    groupId: com.fasterxml.jackson.core
                    artifactId: jackson-core
          """,
        "org.openrewrite.test.RemoveJacksonCore"),
      buildGradle(
        """
          plugins {
              id("java")
          }
          
          def foo() {}
          class A {}
                        
          dependencies {
              implementation("com.fasterxml.jackson.core:jackson-databind:2.15.2")
              implementation("com.fasterxml.jackson.core:jackson-core:2.15.2")
          }
          """,
        """
          plugins {
              id("java")
          }
          
          def foo() {}
          class A {}
                        
          dependencies {
              implementation("com.fasterxml.jackson.core:jackson-databind:2.15.2")
          }
          """
      )
    );
}
```

Yet in rerwite-gradle-plugin the same test now fails:
https://ge.openrewrite.org/s/3qkf6mpikbcjw/tests/task/:plugin:testGradle4/details/org.openrewrite.gradle.RewriteRunTest/gradleDependencyManagement(File)?top-execution=1

**Comment by timtebeek** [comment]
Locally still does not reproduce; so I've gone ahead and remove the GHA caches through
```
gh cache delete --all --repo openrewrite/rewrite-gradle-plugin
```

**Comment by knutwannheden** [comment]
Looks like that didn't help.

**Comment by sambsnyd** [comment]
Fixed in https://github.com/openrewrite/rewrite-gradle-plugin/commit/3a99e56cc199072d70e634e67515211fcbd87d78 

To support managing transitive Gradle dependencies [we added](49b9df53d9d181eb7bfb2efcb4d2880901b2bf57) a reference to `org.gradle.api.artifacts.dsl.DependencyConstraintHandler` to `GradleParser`. That class was introduced in Gradle 4.5, so a ClassNotFound error results. But we already said we don't support Gradle versions earlier than 4.10, so I've updated the build file to use Gradle 4.10 in the compatibility test rather than 4.0.
